js/background_scripts/background.js

```diff
@@ -71,6 +71,67 @@ chrome.runtime.onMessage.addListener(
     }
 );
 
+//Setup Icon Paths
+var notCompatibleIconPath = chrome.runtime.getURL("images/iconNotCompatible.png");
+var compatibleIconPath = chrome.runtime.getURL("images/icon.png");
+var activeTabId = -1;
+var allTabs = {}
+
+//Check if the newly Selected Tab is compatible
+chrome.tabs.onActivated.addListener(function(activeInfo){
+    chrome.tabs.get(activeInfo.tabId, function (tab) {
+        activeTabId = tab.id;
+        //Check if Tab is Compatible
+        checkIfTabIsCompatible(tab.url, tab.id);
+        updateAddOnIcon(allTabs[tab.id].isCompatible);
+    })
+});
+
+//Set needCheck flag to true when Tab Updates and check the updated Tab
+chrome.tabs.onUpdated.addListener(function(tabID, tabChanges, tab){
+    //If the URL didn´t Change there is no need to recheck
+    if(tabChanges.url == null) {
+        return;
+    }
+    allTabs[tabID].needCheck = true;
```

Comment:
Looks like this can be called when the allTabls[tabID] does not exist yet.
I suggest changing this to:

```
if (allTabs[tabID]) {
	allTabs[tabID].needCheck = true;
}
```